src/tracks/playabletrack/wavetrack/ui/WaveClipAdjustBorderHandle.cpp

```diff
@@ -376,34 +397,63 @@ class AdjustBetweenBorders final : public WaveClipAdjustBorderHandle::AdjustPoli
 };
 
 
-HitTestPreview WaveClipAdjustBorderHandle::HitPreview(const AudacityProject*, bool unsafe)
+HitTestPreview WaveClipAdjustBorderHandle::HitPreviewTrim(const AudacityProject*, bool unsafe, bool isLeftBorder)
 {
     static auto disabledCursor =
         ::MakeCursor(wxCURSOR_NO_ENTRY, DisabledCursorXpm, 16, 16);
-    static auto slideCursor =
-        MakeCursor(wxCURSOR_SIZEWE, TimeCursorXpm, 16, 16);
+    static auto trimCursorLeft =
+        MakeCursor(wxCURSOR_SIZEWE, ClipTrimLeftXpm , 16, 16);
+    static auto trimCursorRight =
+        MakeCursor(wxCURSOR_SIZEWE, ClipTrimRightXpm, 16, 16);
     auto message = XO("Click and drag to move clip boundary in time");
 
     return {
-       message,
-       (unsafe
-        ? &*disabledCursor
-        : &*slideCursor)
+        message,
+        (unsafe
+            ? &*disabledCursor
+            : &*(isLeftBorder ? trimCursorLeft : trimCursorRight))
     };
 }
 
+HitTestPreview WaveClipAdjustBorderHandle::HitPreviewStretch(const AudacityProject*, bool unsafe, bool isLeftBorder)
+{
+    static auto disabledCursor =
+        ::MakeCursor(wxCURSOR_NO_ENTRY, DisabledCursorXpm, 16, 16);
+    static auto stretchCursorLeft =
+        MakeCursor(wxCURSOR_SIZEWE, ClipStretchLeftXpm, 16, 16);
+    static auto stretchCursorRight =
+        MakeCursor(wxCURSOR_SIZEWE, ClipStretchRightXpm, 16, 16);
+    auto message = XO("Click and drag to stretch clip");
 
-WaveClipAdjustBorderHandle::WaveClipAdjustBorderHandle(std::unique_ptr<AdjustPolicy>& adjustPolicy)
-   : mAdjustPolicy{ std::move(adjustPolicy) }
+    return {
+        message,
+        (unsafe
+            ? &*disabledCursor
+            : &*(isLeftBorder ? stretchCursorLeft : stretchCursorRight))
+    };
+}
+
+WaveClipAdjustBorderHandle::WaveClipAdjustBorderHandle(std::unique_ptr<AdjustPolicy>& adjustPolicy,
+                                                       bool stretchMode,
+                                                       bool leftBorder)
+    : mAdjustPolicy{ std::move(adjustPolicy) }
+    , mIsStretchMode{stretchMode}
+    , mIsLeftBorder{leftBorder}
 {
 
 }
 
+WaveClipAdjustBorderHandle::~WaveClipAdjustBorderHandle() = default;
+
+WaveClipAdjustBorderHandle::WaveClipAdjustBorderHandle(WaveClipAdjustBorderHandle&&) noexcept = default;
+
+WaveClipAdjustBorderHandle& WaveClipAdjustBorderHandle::operator=(WaveClipAdjustBorderHandle&&) noexcept = default;
+
 void WaveClipAdjustBorderHandle::AdjustPolicy::Draw(TrackPanelDrawingContext&, const wxRect&, unsigned) { }
 
 wxRect WaveClipAdjustBorderHandle::AdjustPolicy::DrawingArea(TrackPanelDrawingContext&, const wxRect& rect, const wxRect&, unsigned)
 {
-   return rect;
+    return rect;
```

Comment:
The previous formatting was the correct one.

Reply:
Unfortunately this file was created with 4-space intedentation, and later few lines got 3-space tab. May be you are right, and it would be better to continue replace 4-space tabs with 3-space tabs line by line, until entire file gets formatted correctly